contracts/plugins/lido/LidoStakedEthCollateral.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: BlueOak-1.0.0
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "contracts/plugins/assets/AppreciatingFiatCollateral.sol";
+import "contracts/plugins/lido/IWSTETH.sol";
+import "contracts/plugins/assets/OracleLib.sol";
+import "contracts/libraries/Fixed.sol";
+
+/**
+ * @title Lido Staked Eth Collateral
+ * @notice Collateral plugin for Lido stETH,
+ * tok = wstETH  (wrapped stETH)
+ * ref = stETH (pegged to ETH 1:1)
+ * tar = ETH
+ * UoA = USD
+ */
+contract LidoStakedEthCollateral is AppreciatingFiatCollateral {
+    using OracleLib for AggregatorV3Interface;
+    using FixLib for uint192;
+
+    AggregatorV3Interface public immutable targetPerRefChainlinkFeed; // {ref/tok}
+    uint48 public immutable targetPerRefChainlinkTimeout; // {s}
+
+    constructor(
+        CollateralConfig memory config,
+        uint192 revenueHiding,
+        AggregatorV3Interface _targetPerRefChainlinkFeed,
+        uint48 _targetPerRefChainlinkTimeout
+    ) AppreciatingFiatCollateral(config, revenueHiding) {
+        require(address(_targetPerRefChainlinkFeed) != address(0), "missing targetPerRef feed");
+        require(_targetPerRefChainlinkTimeout > 0, "targetPerRefChainlinkTimeout zero");
+        targetPerRefChainlinkFeed = _targetPerRefChainlinkFeed;
+        targetPerRefChainlinkTimeout = _targetPerRefChainlinkTimeout;
+        exposedReferencePrice = _underlyingRefPerTok().mul(revenueShowing);
+    }
+
+    /// Can revert, used by other contract functions in order to catch errors
+    /// @param low {UoA/tok} The low price estimate
+    /// @param high {UoA/tok} The high price estimate
+    /// @param pegPrice {target/ref}
+    function tryPrice()
+        external
+        view
+        override
+        returns (
+            uint192 low,
+            uint192 high,
+            uint192 pegPrice
+        )
+    {
+        // {UoA/target}
+        uint192 pricePerTarget = chainlinkFeed.price(oracleTimeout);
```

Comment:
I see 3 chainlink official feeds for StETH. The `StETH / ETH` feed has the widest error of them all. We could consider using the `STETH / USD` feed directly in `tryPrice()`. It would reduce compounding error. Then we could use the `STETH / ETH` feed only in the refresh() function to check for the peg.

Reply:
![Screen Shot 2023-03-17 at 11 24 13 AM](https://user-images.githubusercontent.com/13439795/225948085-c3cc516c-c425-4559-8bbb-69b1b15887a9.png)

see here: https://docs.chain.link/data-feeds/price-feeds/addresses